sklearn/ensemble/losses.py

```diff
@@ -0,0 +1,875 @@
+        """Compute the least squares loss.
+
+        Parameters
+        ----------
+        y : array, shape (n_samples,)
+            True labels
+
+        raw_predictions : array, shape (n_samples, K)
+            The raw_predictions (i.e. values from the tree leaves)
+
+        sample_weight : array-like, shape (n_samples,), optional
+            Sample weights.
+        """
+        if sample_weight is None:
+            return np.mean((y - raw_predictions.ravel()) ** 2.0)
+        else:
+            return (1.0 / sample_weight.sum() * np.sum(
+                sample_weight * ((y - raw_predictions.ravel()) ** 2.0)))
+
+    def negative_gradient(self, y, raw_predictions, **kargs):
+        """Compute the negative gradient.
+
+        Parameters
+        ----------
+        y : array, shape (n_samples,)
+            The target labels.
+
+        raw_predictions : array, shape (n_samples,)
+            The raw_predictions (i.e. values from the tree leaves) of the
+            tree ensemble at iteration ``i - 1``.
+        """
+        return y - raw_predictions.ravel()
+
+    def update_terminal_regions(self, tree, X, y, residual, raw_predictions,
+                                sample_weight, sample_mask,
+                                learning_rate=0.1, k=0):
+        """Least squares does not need to update terminal regions.
+
+        But it has to update the predictions.
+
+        Parameters
+        ----------
+        tree : tree.Tree
+            The tree object.
+        X : array, shape (n, m)
+            The data array.
+        y : array, shape (n,)
+            The target labels.
+        residual : array, shape (n,)
+            The residuals (usually the negative gradient).
+        raw_predictions : array, shape (n_samples, K)
+            The raw_predictions (i.e. values from the tree leaves) of the
+            tree ensemble at iteration ``i - 1``.
+        sample_weight : array, shape (n,)
+            The weight of each sample.
+        sample_mask : array, shape (n,)
+            The sample mask to be used.
+        learning_rate : float, default=0.1
+            learning rate shrinks the contribution of each tree by
+             ``learning_rate``.
+        k : int, default 0
+            The index of the estimator being updated.
+        """
+        # update predictions
+        raw_predictions[:, k] += learning_rate * tree.predict(X).ravel()
+
+    def _update_terminal_region(self, tree, terminal_regions, leaf, X, y,
+                                residual, raw_predictions, sample_weight):
+        pass
+
+    def get_init_raw_predictions(self, X, estimator):
+        return estimator.predict(X).reshape(-1, 1).astype(np.float64)
+
+
+class LeastAbsoluteError(RegressionLossFunction):
+    """Loss function for least absolute deviation (LAD) regression.
+
+    Parameters
+    ----------
+    n_classes : int
+        Number of classes
+    """
+    def init_estimator(self):
+        # Predicts the median
```

Comment:
I think that you can remove this comment. Strategy quantile with 0.5 should be self explicit (and this is the Least Absolute Error) as well